feat(/security/disa-stig): Page redesign

petesfrench opened the pull request:
## Done

- Redesign [/security/dis-stig](https://ubuntu-com-14702.demos.haus/security/disa-stig) to match the [figma design](https://www.figma.com/design/pc4AcZqcXvWJqrm4t1x1y6/ubuntu.com%2Fsecurity?node-id=911-5131&t=KfUtcoGP63SEgk98-0)
- Some minor content/spelling changes 
- Created `no-full-stop` modifier class for use on `p-stepped-list`

## QA

- Open [the demo](https://ubuntu-com-14702.demos.haus/security/disa-stig)
- Compare against the [figma design](https://www.figma.com/design/pc4AcZqcXvWJqrm4t1x1y6/ubuntu.com%2Fsecurity?node-id=911-5131&t=KfUtcoGP63SEgk98-0)
- Compare against the [copydoc](https://docs.google.com/document/d/1zJwZzc-cERj9YKNXFtmrXwynJptyT7-D3qCNzBctRfo/edit?tab=t.0)
- Check the page on all screen sizes

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-12037

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14702.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/13441)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

petesfrench commented:
@mattea-turic @eliman11 This one is ready for review

eliman11 commented:
Looks pretty much perfect, thanks @petesfrench! Just one comment - can you change the background to our paper color (#F3F3F3)? Will +1 anyways

mattea-turic commented:
Thank you @petesfrench ! 

For the hero:
- [ ] I forgot to add a break after "Security" in the figma, so could you add this in pls?
- [ ] The disa logo is blurry; not sure why, since the [asset](https://assets.ubuntu.com/v1/92754de5-disa.png) looks sharp enough

For 'DISA-STIG for Ubuntu":
- [ ] Could you apply the formatting for the table that Elaine added in Figma? We weren't really sure which to go for, but seeing the current table on mobile, it would probably make more sense to use that one so the user won't have to scroll so much. I hid the old one, so it should be clear which one I'm talking about

For 'How to audit and comply with DISA-STIG?"
- [ ] Could the list numbers sit at the start of the HR/7th column? Right now there's some excess space there 
<img width="500" alt="Screenshot 2025-02-03 at 12 38 42" src="https://github.com/user-attachments/assets/3fe99f14-11b7-4bc5-8a12-2884e51ee885" />
<br>
<br>
For "Resources":
- [ ] There should be deep bottom padding

petesfrench commented:
Thanks @mattea-turic, I have addressed all your comments.

One note on this one:

They have the counter text [has styling to align the text to the right](https://github.com/canonical/vanilla-framework/blob/a2b776370a1d9ea1d9076876bc50e823268c8640/scss/_patterns_lists.scss#L37). Not sure if this has been done for a reason

mattea-turic commented:
Perfect, thanks @petesfrench ! I'll +1

petesfrench commented:
@anthonydillon This one is ready for code&qa

anthonydillon commented on static/sass/styles.scss:
```diff
@@ -1766,3 +1766,11 @@ legend {
   }
 }
+
+// XXX: Custom styling to remove the full-stop after the counter in 'p-stepped-list'
```
Could you create an issue on vanilla to purpose this change please?

petesfrench replied:
PR created [here](https://github.com/canonical/vanilla-framework/pull/5465)

anthonydillon commented on templates/security/disa-stig.html:
```diff
@@ -1,190 +1,295 @@
+    {%- endif -%}
+    {%- if slot == 'image' -%}
+      <div class="p-image-container--cinematic is-cover">
+        {{ image(url="https://assets.ubuntu.com/v1/f759a607-hero.png",
```
This is a very large PNG. Could we try with fmt=jpg to reduce the size without losing to much data?

petesfrench replied:
AKAIK the [image template](https://github.com/canonical/canonicalwebteam.image-template/tree/main) is not able to take format as a parameter, I have made an [issue here to add this functionality](https://warthogs.atlassian.net/browse/WD-18977).

In the meantime, could we try a JPEG version of some of the bigger images on this page? @mattea-turic

anthonydillon commented on templates/security/disa-stig.html:
```diff
@@ -1,190 +1,295 @@
-        The Defense Information System Agency (DISA) is a US Department of Defense combat support agency. It provides and operates information infrastructure to support military operations and national-level leadership. The Security Technical Implementation Guide (STIG) is a configuration standard consisting of guidelines for hardening systems to improve a system’s security posture. It can be see ...
-      </p>
+      <div class="p-image-container--cinematic is-cover">
+        {{ image(url="https://assets.ubuntu.com/v1/3c7382e4-what-is-disa.png",
```
Same here. A jpeg would save a lot of page weight

anthonydillon reviewed (approved):
Looks good Pete. There are a few suggestions in the code. There are several large PNGs which will make the total page size rather large by the time the user reaches the end. Could you try and tweak some formatting to get the best result without compromising much on design.

petesfrench commented:
@mattea-turic Should be working now

mattea-turic commented:
thanks @petesfrench! 

Looks good apart from the webinar now having a background – if you've re-exported this as a jpg, then that doesn't support transparency so perhaps maintain png for this one?
